scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py

```diff
@@ -0,0 +1,880 @@
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Settings
+    # ------------------------------------------------------------------------------------------------------------------
+    def _setup_instrument_specific_settings(self):
+        # Get the first run number of the scatter data for the first table
+        sample_scatter = self._view.get_cell(row=0, column=0, convert_to=str)
+
+        # Check if it exists at all
+        if not sample_scatter:
+            return
+
+        # Get the file information from
+        file_information_factory = SANSFileInformationFactory()
+        try:
+            self._file_information = file_information_factory.create_sans_file_information(sample_scatter)
+        except NotImplementedError:
+            sans_logger.warning("Could not get file information from {}.".format(sample_scatter))
+            self._file_information = None
+
+        # Provide the instrument specific settings
+        if self._file_information:
+            # Set the instrument on the table
+            instrument = self._file_information.get_instrument()
+            self._view.set_instrument_settings(instrument)
+
+            # Set the reduction mode
+            reduction_mode_list = get_reduction_mode_strings_for_gui(instrument=instrument)
+            self._view.set_reduction_modes(reduction_mode_list)
+        else:
+            self._view.set_instrument_settings(SANSInstrument.NoInstrument)
+            reduction_mode_list = get_reduction_mode_strings_for_gui()
+            self._view.set_reduction_modes(reduction_mode_list)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Setting workaround for state in DataProcessorWidget
+    # ------------------------------------------------------------------------------------------------------------------
+    def _remove_dummy_workspaces_and_row_index(self):
+        number_of_rows = self._view.get_number_of_rows()
+        for row in range(number_of_rows):
+            self._remove_from_hidden_options(row, "InputWorkspace")
+            self._remove_from_hidden_options(row, "RowIndex")
+
+    def _set_indices(self):
+        number_of_rows = self._view.get_number_of_rows()
+        for row in range(number_of_rows):
+            to_set = Property.EMPTY_INT if self.is_empty_row(row) else row
+            self._add_to_hidden_options(row, "RowIndex", to_set)
+
+    def _set_dummy_workspace(self):
+        number_of_rows = self._view.get_number_of_rows()
+        for row in range(number_of_rows):
+            self._add_to_hidden_options(row, "InputWorkspace", SANS_DUMMY_INPUT_ALGORITHM_PROPERTY_NAME)
+
+    @staticmethod
+    def _create_dummy_input_workspace():
+        if not AnalysisDataService.doesExist(SANS_DUMMY_INPUT_ALGORITHM_PROPERTY_NAME):
+            create_alg = AlgorithmManager.create("CreateSampleWorkspace")
```

Comment:
If you just need a workspace with "nothing" in it could we just create a 1 spectra/1 bin workspace using the workspace factory and add it to the ADS manually? It would save a lot of work here.